project/driver_controls/inc/driver_device.h

```diff
@@ -0,0 +1,25 @@
+#pragma once
+
+// Provides an interface to configure the GPIO pins for the driver control devices.
+// This allows a cleaner device initialization than having to initialize each pin with
+// different settings
+
+#include "gpio.h"
+#include "interrupt.h"
+
+typedef void (*DriverDeviceCallback)(GPIOAddress *address, void *context);
+
+typedef struct DriverDevice {
```

Comment:
I'm not entirely convinced you need to expose all these options. At this point you might as well just ask them to fill out the GPIO setting struct directly. :P